packages/gatsby-plugin-styled-components/src/gatsby-ssr.js

```diff
@@ -11,8 +11,8 @@ exports.replaceServerBodyRender = ({ component }) => {
     </StyleSheetManager>
   )
 
-  const html = renderToString(app)
-  const styles = sheet.getStyleElement()
+  const body = renderToString(app)
+  const headComponents = sheet.getStyleElement()
```

Comment:
`headComponents` is already passed in as an array. You just need to push onto it the new element from Styled Components.